Classes/Replicator/NodeReplicator.php

```diff
@@ -138,4 +145,34 @@ protected function logReplicationAction(NodeInterface $nodeVariant, string $mess
 
         $this->logger->log($logLevel, sprintf('[NodeIdentifier: %s, TargetDimension: %s] %s', $nodeVariant->getIdentifier(), $dimensionString, $message), LogEnvironment::fromMethodName($loggingMethod ?? __METHOD__));
     }
+
+    /**
+     * @param NodeInterface $nodeVariant
+     * @param array|null $excludedProperties
+     * @return array
+     * @throws NodeException
+     */
+    protected function getExcludedPropertyValues(NodeInterface $nodeVariant, ?array $excludedProperties): array
+    {
+        $excludedPropertyValues = [];
+        if ($excludedProperties) {
+            foreach ($excludedProperties as $property) {
+                $excludedPropertyValues[$property] = $nodeVariant->getProperty($property);
+            }
+        }
+        return $excludedPropertyValues;
+    }
+
+    /**
+     * @param NodeInterface $nodeVariant
+     * @param array $excludedPropertyValues
+     */
+    protected function setExcludedPropertyValues(NodeInterface $nodeVariant, array $excludedPropertyValues): void
+    {
+        if ($excludedPropertyValues) {
```

Comment:
$excludedPropertyValues is neither `false` nor `true`, please use empty($excludedPropertyValues). An early return would reduce code complexity.